starknet_py/contract.py

```diff
@@ -611,12 +648,15 @@ def __init__(
         client: GatewayClient,
         max_steps: int,
         proxy_checks: List[ProxyCheck],
+        account: Optional[BaseAccount] = None,
```

Comment:
Shouldn't we use `_unpack_client_and_account` to get Account from AccountClient here and create mutable contract in _make_contract_recursively?